### DISCUSSION TITLE
Need more help about migrate

### DISCUSSION THREAD

**Discussion by MedLabs** [pocketbase/pocketbase#1942] [Q&A]
Hi,
Each time there is a new version, I download it then I run pocketbase migrate :/ and lose my collections in some pb instances...

I have multiple pb_data-folders (pb_data, pb_user1, pb_user2, pb_user3...)
**pb_data** is different from the others (it's kinda dashboard for users management)
the others have the same schema.

when I run migrate ( now I know that I shouldn't run it each PB new version ) my users dbs structures are refactored like the pb_data schema.
is **pb_data** is used as default structure ?
Why pocketbase migrate changes my other db schema to the pb_data schema even if I never made a migration script ?
Mybe we should have a more clear confirmation message when we run migrate ?

Please explain to me the migrate thing that copy pb_data schema to other folders.
Thanks

**Comment by ganigeorgiev** [comment]
How are you starting the different PocketBase applications?
Are all pb_data within the same parent directory?
Do you have a `pb_migrations` directory?

If you use a single `pb_migrations` dir for all of your apps, then all apps will have the same migrations. You can change the default migrations directory with the `--migrationsDir` flag (by default the migrations dir is created relative to the `pb_data` dir). You can find information for all available flags with the `--help`.

Could you provide a little more info about your setup or at least reproducible steps?

**Reply by MedLabs** [reply]
Ok, I'm using ./pocketbase folder, and all my users dbs have subfolders (/pocketbase/pb_data, /pocketbase/pb_user1, /pocketbase/pb_user2 ...) and there is a ./pocketbase/pb_migrations folder amoung all folders.
pb_data/ contains a pb instance of the main dashboard to manage users servers_urls ...
pb_userX contains the user's db files

when I overwrite the pocketbase executable I run "pocketbase migrate", and you know what ? pb_user1 schema is overwritten (to pb_data schema) but pb_user2 is not :laughing:  I don't know why.

my instances are added to pm2 so to run them all I use pm2 start all.

----
So you're saying that I should add --migrationDir="migramtion_userX" on every pocketbase serve command ?

**Reply by ganigeorgiev** [reply]
Why do you run `pocketbase migrate` in the first place?

Note that everytime when you execute a `pocketbase` command it is associated to a `pb_data` dir. For example `pocketbase migrate` is the same as `pocketbase migrate --dir='./pb_data'`.

If you want to have a single executable running multiple `pb_data` directories each with different schema and migrations, then you need to specify both `--dir` and `--migrationsDir` OR wrap each `pb_data` in its own directory because by default pb_migrations is relative next to the `pb_data` location, for example:

```sh
pocketbase/
  pocketbase            # the PocketBase executable
  pb1/                  # ./pocketbase serve --dir='./pb1/pb_data' --http='127.0.0.1:8091'
     pb_data/
     pb_migrations/
  pb2/                  # ./pocketbase serve --dir='./pb2/pb_data' --http='127.0.0.1:8092'
     pb_data/
     pb_migrations/
  pb3/                  # ./pocketbase serve --dir='./pb3/pb_data' --http='127.0.0.1:8093'
     pb_data/
     pb_migrations/
```

**Reply by MedLabs** [reply]
since 0.8.0 I ran update then I thought that migrate will update db schemas to update breaking changes...
Now I got it: no need for migrate command if I did not add any custom migration files to my db schemas.

Thanks a lot !

**Reply by ganigeorgiev** [reply]
You don't have to run `./pocketbase migrate up` manually even if you create custom migrations.

All unapplied migrations are executed automatically before starting the server (aka. on `./pocketbase serve`).
